src/framework/components/element/text-element.js

```diff
@@ -493,6 +505,17 @@ Object.assign(TextElement.prototype, {
             this._element.addModelToLayers(this._model);
         }
 
+        // Remove the non visible symbols
+        var inputIndex = 0;
+        for (i = 0; i < this._symbols.length; i++) {
+            if (!NON_VISIBLE_CHAR.has(this._symbols[i])) {
+                this._symbols[inputIndex] = this._symbols[i];
+                inputIndex++;
+            }
+        }
```

Comment:
Am I missing something here? The `this._symbols` array is created empty. This `for` loop iterates it but it has zero elements. And elements are only added inside. And `inputIndex` is only non-zero if code inside the loop is executed. I can't see how `this._symbols` is ever anything other than the empty array. This might be a 'brain still not in gear in the morning' moment on my part.

Reply:
`this._symbols` is populated at the top of the function: https://github.com/playcanvas/engine/blob/master/src/framework/components/element/text-element.js#L262 :)

Reply:
Oh right, so you're effectively throwing away the initially assigned empty array and overwriting the property with a new array. Might as well initially assign it as null perhaps?

Reply:
I've should have explained what the bug was in the PR 😅. My bad!

One of our clients has a RTL localisation text that also has these 'markers' which changes a section of the text be LTR

Unfortunately, these markers get rendered as empty squares:
![image](https://user-images.githubusercontent.com/16639049/96268168-a1587b00-0fc0-11eb-9c0e-3f9f708edbbc.png)

Before this loop executes, `_symbols` contains an array of each character in the text including the markers. This loop is an attempt to remove the markers without allocating a new array and save on garbage collection.

Reply:
OK. Well, this doesn't look too bad performance wise. Any reasons/concerns that'd make you hesitate to merge this?

Reply:
actually, I think symbols should be removed before the call to _calculateCharsPerTexture().

also, this will probably break text markup as symbolColors needs updating.